libnd4j/include/ops/declarable/platform/armcompute/armcomputeUtils.cpp

```
@@ -0,0 +1,278 @@
+  return tensor;
+}
+
+void copyFromTensor(const Arm_Tensor& inTensor, NDArray& output) {
+    //only for C order
+    //only for C order
+    if (output.ordering() != 'c') return;
+    auto shapeInfo = output.shapeInfo();
+    auto bases = &(shapeInfo[1]);
+    Nd4jLong rank = shapeInfo[0];
+    auto strides = output.stridesOf();
+    int width = bases[rank - 1];
+    uint8_t* outputBuffer = (uint8_t*)output.buffer(); 
+    size_t offset = 0;
+    arm_compute::Window window;
+    arm_compute::Iterator tensor_it(&inTensor, window);
+
+    int element_size = inTensor.info()->element_size();
+    window.use_tensor_dimensions(inTensor.info()->tensor_shape(), /* first_dimension =*/arm_compute::Window::DimY);
+
+//    if (output.ews() == 1) {
+        auto copySize = width * element_size;
+        auto dest = outputBuffer;
+        arm_compute::execute_window_loop(window, [&](const arm_compute::Coordinates& id)
+            {
+                auto src = tensor_it.ptr(); 
+                memcpy(dest, src, copySize);
+                dest += copySize;
+            },
+            tensor_it);
+    // }
+    // else {
+    //     Nd4jLong coords[MAX_RANK] = {};
+    //     if(strides[rank-1]!=1){
+    //             throw std::runtime_error(  "not implemented for subarrays whose last stride is not 1");
+    //         //TODO: implement to work with all subarrays properly
+    //     }
+    //     arm_compute::execute_window_loop(window, [&](const arm_compute::Coordinates& id)
+    //         {
+    //             auto src = tensor_it.ptr();
+    //             auto dest = outputBuffer + offset * element_size;
+    //             memcpy(dest, src, width * element_size);
+    //             offset = sd::inc_coords(bases, strides, coords, offset, rank, 1);
+    //         },
+    //         tensor_it);
+    // }
+}
+
+void copyToTensor(const NDArray& input, Arm_Tensor& outTensor) {
+    //only for C order
+    if (input.ordering() != 'c') return;
```

Maybe throw an exception here?

---

noted it. I will visit these functions again as armCompute lib support is on progress.